### PR TITLE
feat(destination): Filter the duplicate primary keys prior to writing batch

### PR DESCRIPTION
**Pull request by candiduslynx**
Implements #627

### CODE REVIEW

**Review by hermanschaaf — APPROVED** [review]
LGTM, just minor comments around naming

File: internal/pk/pk.go [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package pk
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/cloudquery/plugin-sdk/schema"
+)
+
+func Convert(table *schema.Table, resource []any) string {
```
Comment (disq) [inline internal/pk/pk.go:10]: Could have a better name (with a godoc) maybe `Stringify`? Also why not get `schema.Resource` as the single param, and use `ToDestinationResource`?

```suggestion
func Stringify(r *schema.Resource) string {
```

Comment (candiduslynx) [reply internal/pk/pk.go:10]: we are working with the data from destination side, so we have table + `[][]any`

[the remark this answers is not in the document]
Comment (candiduslynx) [reply internal/pk/pk.go:10]: done

File: plugins/destination/managed_writer.go [hunk 2]
```diff
@@ -66,6 +71,46 @@ func (p *Plugin) flush(ctx context.Context, metrics *Metrics, table *schema.Tabl
 	}
 }
 
+func (p *Plugin) filterByPK(table *schema.Table, resources [][]any) [][]any {
```
[the remark this answers is not in the document]
Comment (candiduslynx) [reply plugins/destination/managed_writer.go]: done

**Review by yevgenypats — APPROVED** [review]

**Pull request merged**